core/trino-main/src/test/java/io/trino/sql/query/TestRowFilter.java

```diff
@@ -98,8 +102,17 @@ public void init()
                     if (schemaTableName.equals(new SchemaTableName("tiny", "nation_with_hidden_column"))) {
                         return TPCH_WITH_HIDDEN_COLUMN_DATA;
                     }
+                    if (schemaTableName.equals(new SchemaTableName("tiny", "nation_with_optional_column"))) {
+                        return TPCH_NATION_DATA;
+                    }
                     throw new UnsupportedOperationException();
                 })
+                .withAllowMissingColumnsOnInsert(schemaTableName -> {
```

Comment:
Instead of modyfying SPI, I think it would be better to setup another catalog in this test with a different MockConnectorFactory setup.